ci-operator/step-registry/upi/install/azure/upi-install-azure-commands.sh

```diff
@@ -180,13 +199,19 @@ az role assignment create --assignee "$PRINCIPAL_ID" --role 'Contributor' --scop
 
 pushd /tmp/azure
 
-echo "Deploying 01_vnet"
-az deployment group create -g $RESOURCE_GROUP \
-  --template-file "01_vnet.json" \
-  --parameters baseName="$INFRA_ID"
+if [ -v vnet_name ] && [ -n "${vnet_name}" ]; then
+  echo "Using the existing existing ${vnet_name}"
+  echo "Linking VNet to private DNS zone"
+  az network private-dns link vnet create -g $RESOURCE_GROUP -z ${CLUSTER_NAME}.${BASE_DOMAIN} -n ${INFRA_ID}-network-link -v "${vnet_name}" -e false
+else
+  echo "Deploying 01_vnet"
+  az deployment group create -g $RESOURCE_GROUP \
+    --template-file "01_vnet.json" \
+    --parameters baseName="$INFRA_ID"
 
-echo "Linking VNet to private DNS zone"
-az network private-dns link vnet create -g $RESOURCE_GROUP -z ${CLUSTER_NAME}.${BASE_DOMAIN} -n ${INFRA_ID}-network-link -v "${INFRA_ID}-vnet" -e false
+  echo "Linking VNet to private DNS zone"
+  az network private-dns link vnet create -g $RESOURCE_GROUP -z ${CLUSTER_NAME}.${BASE_DOMAIN} -n ${INFRA_ID}-network-link -v "${INFRA_ID}-vnet" -e false
```

Comment:
We don't want to repeat this command in different if/else blocks. Instead, write the command once and use conditionals to change the values that are passed in.

Reply:
Done